dspy/clients/lm.py

```diff
@@ -95,8 +95,23 @@ def __call__(self, prompt=None, messages=None, **kwargs):
             request=ujson.dumps(dict(model=self.model, messages=messages, **kwargs)),
             num_retries=self.num_retries,
         )
-        outputs = [c.message.content if hasattr(c, "message") else c["text"] for c in response["choices"]]
-
+        if self.kwargs.get("logprobs"):
+            outputs = [
+                {
+                    "text": c.message.content if hasattr(c, "message") else c["text"],
+                    "logprobs": c.logprobs if hasattr(c, "logprobs") else c["logprobs"]
+                }
+                for c in response["choices"]
+            ]
+        else:
+            outputs = [
+                {
+                    "text": c.message.content if hasattr(c, "message") else c["text"],
```

Comment:
This appears to change the return type of `__call__` even if the user doesn't request logprobs? We normally return a list of strings. This seems to return a list of dicts even if logprobs=False.